### PR TITLE
Clean up legacy v1 header logic, use built-in Base64 since JDK8

### PR DESCRIPTION
**Pull request by kezhenxu94**
### Motivation:

Codes clean up

### Modifications:

- Remove v1 header, follow up #4244

- Use built-in Base64 class (since JDK8)

### Result:

- No more legacy v1 headers

- No unnecessary codes concerns by using built-in ability

### CODE REVIEW

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java [hunk 1]
```diff
@@ -108,53 +103,36 @@ String serialize(HeaderVersion version) {
...
+                    this.primaryDistributedTraceId = new PropagatedTraceId(Base64.decode2UTFString(parts[1]));
+                    this.traceSegmentId = new ID(Base64.decode2UTFString(parts[2]));
+                    this.spanId = Integer.parseInt(parts[3]);
+                    this.parentServiceInstanceId = Integer.parseInt(parts[4]);
+                    this.entryServiceInstanceId = Integer.parseInt(parts[5]);
+                    this.peerHost = Base64.decode2UTFString(parts[6]);
+                    this.entryEndpointName = Base64.decode2UTFString(parts[7]);
+                    this.parentEndpointName = Base64.decode2UTFString(parts[8]);
+                } catch (NumberFormatException ignored) {
+
```
Comment (kezhenxu94) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java]: Quick return to reduce nested levels

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java [hunk 2]
```diff
@@ -108,53 +103,36 @@ String serialize(HeaderVersion version) {
...
                 }
-            } else {
-                throw new IllegalArgumentException("Unimplemented header version." + version);
             }
         }
-        return this;
+        throw new IllegalArgumentException("Unimplemented header version." + version);
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java]: Don't throw an exception in agent core API. Output a warning log.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java [hunk 3]
```diff
@@ -163,27 +141,16 @@ public boolean isValid() {
...
                 && traceSegmentId.isValid()
                 && getSpanId() > -1
                 && parentServiceInstanceId != DictionaryUtil.nullValue()
                 && entryServiceInstanceId != DictionaryUtil.nullValue()
                 && !StringUtil.isEmpty(peerHost)
                 && primaryDistributedTraceId != null;
-        } else {
-            throw new IllegalArgumentException("Unimplemented header version." + version);
         }
+        throw new IllegalArgumentException("Unimplemented header version." + version);
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java]: Same here.

**Comment by wu-sheng** [comment]
Many CI processes fail. Please recheck.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java (hunk 2 above)
Comment (kezhenxu94) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextCarrier.java]: Just double check the original codes, the exception statement should never be reached, I simply remove them

**Pull request merged**